### PR TITLE
SIM: Pedometer and YAML config for job runner

### PR DESCRIPTION
**Pull request by mattlangford**

### CODE REVIEW

File: simulation/sub8_gazebo/diagnostics/job_runner.py [hunk 1]
```diff
@@ -113,22 +90,59 @@ def start_caching(self):
...
+        self.cache_dict = {}
+        msgs = messages_to_bag['MESSAGES']
+        for _, msg in msgs.items():
+            # Get message information
+            msg_topic = msg['message_topic']
+            msg_type = msg['message_type']
+            msg_name = msg['message_name']
+
+            # Import the message
+            exec("from {0}.msg import {1}".format(msg_type, msg_name))
```
Comment (mattlangford) [inline simulation/sub8_gazebo/diagnostics/job_runner.py:114]: Python is magical

File: simulation/sub8_gazebo/nodes/gazebo_controller.py [hunk 2]
```diff
@@ -104,6 +130,10 @@ def publish_odom(self, *args):
                     twist=twist
                 )
             )
+
+            dist = np.linalg.norm(sub8_utils.twist_to_numpy(twist)) * self.odom_freq
```
Comment (mattlangford) [inline simulation/sub8_gazebo/nodes/gazebo_controller.py:134]: Probably should do a real frequency calculation here but whatever

File: simulation/sub8_gazebo/diagnostics/job_runner.py [hunk 3]
```diff
@@ -113,22 +90,59 @@ def start_caching(self):
         self.dump()
 
     @txros.util.cancellableInlineCallbacks
-    def dump(self, post_record_time=1):
+    def make_dict(self):
+        '''
+        Generate caching dictionary from a yaml file.
+        '''
+        with open(DIAG_OUT_URI + 'messages_to_bag.yaml', 'r') as f:
```
Comment (jpanikulam) [inline simulation/sub8_gazebo/diagnostics/job_runner.py]: Heade up, URI might not be the word you're looking for

File: simulation/sub8_gazebo/diagnostics/job_runner.py [hunk 4]
```diff
@@ -113,22 +90,59 @@ def start_caching(self):
...
+            messages_to_bag = yaml.load(f)
+
+        # Set bagging parameters
+        self.time_step = messages_to_bag['PARAMS']['time_step']
+        self.pre_cache_time = messages_to_bag['PARAMS']['pre_cache_time']
+        self.post_cache_time = messages_to_bag['PARAMS']['post_cache_time']
+
+        self.cache_dict = {}
+        msgs = messages_to_bag['MESSAGES']
+        for _, msg in msgs.items():
```
Comment (jpanikulam) [inline simulation/sub8_gazebo/diagnostics/job_runner.py]: Fyi msg.values() does this

**Comment by jpanikulam** [comment]
Looks like there's a missing setup.py causing make to fail

**Comment by mattlangford** [comment]
I removed the one I put in there acouple commits ago. Does the python line from CMake file need to be removed too?

**Comment by jpanikulam** [comment]
Oh yup yes. Setup_python tells catkin to run the setup.py
- Jake

**Comment by jpanikulam** [comment]
+1 for the amend

**Pull request merged**